index.test-d.ts

```diff
@@ -1,18 +1,19 @@
-import {expectType, expectAssignable} from 'tsd';
+import {expectTypeOf} from 'expect-type';
 import {getProperty, setProperty, hasProperty, deleteProperty} from './index.js';
 
-expectType<string>(getProperty({foo: {bar: 'unicorn'}}, 'foo.bar'));
-expectType<undefined>(getProperty({foo: {bar: 'a'}}, 'foo.notDefined.deep'));
-expectAssignable<string>(
+expectTypeOf(getProperty({foo: {bar: 'unicorn'}}, 'foo.bar')).toBeString();
+expectTypeOf(getProperty({foo: {bar: 'a'}}, 'foo.notDefined.deep')).toBeUndefined();
+expectTypeOf(
 	getProperty({foo: {bar: 'a'}}, 'foo.notDefined.deep', 'default value'),
-);
-expectType<string>(
+).toBeString();
+expectTypeOf(
 	getProperty({foo: {'dot.dot': 'unicorn'}}, 'foo.dot\\.dot'),
-);
+	// @ts-expect-error type-fest's `Get` not smart enough to deal with escaped dots
+).toEqualTypeOf<string>();
```

Comment:
@sindresorhus worth noting - when unsure what the return type would be, it defaults to `undefined`. The "default" in `type-fest.Get` is `unknown`, since typescript in general allows for additional properties, so we can't know for sure it's `undefined`. It seems that `getProperty` effectively overrides the behaviour by making the default `DefaultValue` undefined - IMO it shouldn't. I want to keep this PR focused so don't want to change that here but I think it should be considered to switch to `unknown` (obviously best solution is to make `Get` respect `\.` but that would make it incompatible with `lodash.get` so 🤷

Reply:
I agree: https://github.com/sindresorhus/dot-prop/issues/95